specifyweb/frontend/js_src/lib/components/QueryBuilder/Line.tsx

```diff
@@ -230,6 +232,11 @@ export function QueryLine({
     availableFilters.length > 1 || availableFilters[0][0] !== 'any';
 
   const hasAny = field.filters.some(({ type }) => type === 'any');
+  const [collapse, setCollapse] = React.useState(true);
+  React.useEffect(() => {
+    if (isAllCollapsed === undefined) return;
+    setCollapse(isAllCollapsed);
+  }, [isAllCollapsed]);
```

Comment:
A shorter way to write this logic:
```
const [collapse, setCollapse] = useTriggerState(isAllCollapsed);
```

Take a look at the description of that hook to see what it does. If you have time, please also look over other hooks in that directory as they all are quite useful in some cases